time/src/parsing/parsable.rs

```diff
@@ -520,9 +520,20 @@ impl sealed::Sealed for Rfc3339 {
         let input = exactly_n_digits::<2, _>(input)
             .and_then(|item| item.consume_value(|value| parsed.set_day(value)))
             .ok_or(InvalidComponent("day"))?;
-        let input = ascii_char_ignore_case::<b'T'>(input)
-            .ok_or(InvalidLiteral)?
-            .into_inner();
+
+        // RFC3339 allows any separator, not just `T`, not just `space`.
+        // cf. Section 5.6: Internet Date/Time Format:
+        //   NOTE: ISO 8601 defines date and time separated by "T".
+        //   Applications using this syntax may choose, for the sake of
+        //   readability, to specify a full-date and full-time separated by
+        //   (say) a space character.
+        // Specifically, rusqlite uses space separators.
+        let input = if input.is_empty() {
+            return Err(InvalidComponent("separator").into());
+        } else {
+            &input[1..]
```

Comment:
Oddly enough, I neither documented this anywhere nor enabled the relevant clippy lints for the module, but generally speaking I try to avoid slicing even if it's obvious to a human that it cannot panic (as is the case here). While the compiler is likely able to elide the panic, the point at which it does so during compilation is past the point where it determines what is easily inlined. So this _could_ have an actual effect on performance even if it's elided.

This seems super nitpicky, but I would prefer to have it written in another manner so as to avoid slicing of the panicking variety. If you'd rather I do this, not an issue! I'll get this merged quite soon in any situation.

Reply:
I refactored it to use `.get()` instead, which shouldn't panic and takes care of the empty case as well. it's even more readable!

Reply:
That's even better than I expected it to look. The same thing exists on line 641 FYI.

Reply:
Fixed line 641 and squashed all my commits into one